src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -14,211 +24,3020 @@ namespace Microsoft.CodeAnalysis.CSharp
+                    {
+                        var backingField = (access.PropertySymbol as SourcePropertySymbol)?.BackingField;
+                        if (backingField != null)
+                        {
+                            VisitFieldAccessInternal(access.ReceiverOpt, backingField);
+                            break;
+                        }
+                    }
+
+                    goto default;
+
+                case BoundKind.FieldAccess:
+                    {
+                        BoundFieldAccess node1 = (BoundFieldAccess)node;
+                        VisitFieldAccessInternal(node1.ReceiverOpt, node1.FieldSymbol);
+                        break;
+                    }
+
+                case BoundKind.EventAccess:
+                    {
+                        BoundEventAccess node1 = (BoundEventAccess)node;
+                        VisitFieldAccessInternal(node1.ReceiverOpt, node1.EventSymbol.AssociatedField);
+                        break;
+                    }
+
+                case BoundKind.TupleLiteral:
+                    ((BoundTupleExpression)node).VisitAllElements((x, self) => self.VisitLvalue(x), this);
+                    break;
+
+                case BoundKind.SuppressNullableWarningExpression:
+                    VisitLvalue(((BoundSuppressNullableWarningExpression)node).Expression);
+                    break;
+
+                default:
+                    VisitRvalue(node);
+                    break;
             }
 
-            if (catchBlock.ExceptionFilterOpt != null)
+            if (_trackRegions && node == this.lastInRegion && this.regionPlace == RegionPlace.Inside) LeaveRegion();
+        }
+
+        protected virtual void VisitLvalue(BoundLocal node)
+        {
+            // no need for it to be previously assigned: it is on the left.
+        }
+
+        /// <summary>
+        /// Visit a boolean condition expression.
+        /// </summary>
+        /// <param name="node"></param>
+        protected void VisitCondition(BoundExpression node)
+        {
+            Visit(node);
+            AdjustConditionalState(node);
+        }
+
+        private void AdjustConditionalState(BoundExpression node)
+        {
+            if (IsConstantTrue(node))
             {
-                VisitCondition(catchBlock.ExceptionFilterOpt);
-                SetState(StateWhenTrue);
+                Unsplit();
+                SetConditionalState(this.State, UnreachableState());
+            }
+            else if (IsConstantFalse(node))
+            {
+                Unsplit();
+                SetConditionalState(UnreachableState(), this.State);
+            }
+            else if ((object)node.Type == null || node.Type.SpecialType != SpecialType.System_Boolean)
+            {
+                // a dynamic type or a type with operator true/false
+                Unsplit();
             }
 
-            VisitStatement(catchBlock.Body);
+            Split();
```

Comment:
this method could use some commenting.

Reply:
to me, it would be clearer if all four cases were handled entirely independently., with docs for each.